numpy/core/tests/test_multiarray.py

```diff
@@ -3769,10 +3769,15 @@ def test_roundtrip(self):
                                                    ('c', float)])
             ]
 
+            refs = [weakref.ref(a) for a in DATA]
             for a in DATA:
                 assert_equal(
                         a, pickle.loads(pickle.dumps(a, protocol=proto)),
                         err_msg="%r" % a)
+            del a, DATA, carray
+            # check for reference leaks (gh-12793)
+            for ref in refs:
+                assert ref() is None
```

Comment:
@mattip, this type of test will fail on pypy, right?

Reply:
Hmm, yes, I should call `gc.collect`.

Reply:
Could skip for pypy. These sorts of tests do tend to be fragile.